File: CameraSecurityAndroid/CameraSecurityAndroid/CameraSecurityAndroid/Communication/PacketData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CameraSecurityAndroid.Communication
{
    public class PacketData
    {
        private readonly Dictionary<string, string> _attributes;

        public PacketData()
        {
            _attributes = new Dictionary<string, string>();
        }

        public void AddAttribute(string attributeName, string attributeData)
        {
            _attributes.Add(attributeName, attributeData);
        }

        public string GetAttribute(string attributeName)
        {
            if (_attributes.TryGetValue(attributeName, out string val))
            {
                return val;
            }
            return null;
        }

        public IEnumerable<KeyValuePair<string, string>> GetAllAttributes()
        {
            return _attributes.AsEnumerable();
        }

        public bool IsValid()
        {
            if (_attributes.TryGetValue("code", out string code))
            {
                if (int.TryParse(code, out _))
                {
                    return true;
                }
            }
            return false;
        }
    }
}

File: CameraSecurity/app/src/main/java/com/lzaromskis/camerasecurity/communication/PacketData.java
package com.lzaromskis.camerasecurity.communication;

import android.util.Pair;

import java.util.Dictionary;
import java.util.Enumeration;
import java.util.Hashtable;

public class PacketData {
    private final Dictionary<String, String> _attributes;

    public PacketData() {
        _attributes = new Hashtable<>();
    }

    public void addAttribute(String attributeName, String attributeValue) {
        _attributes.put(attributeName, attributeValue);
    }

    public String getAttribute(String attributeName) {
        return _attributes.get(attributeName);
    }

    public Enumeration<String> getAllAttributeNames() {
        return _attributes.keys();
    }

    public Boolean isValid() {
        String val = _attributes.get("code");
        if (val != null)
        {
            try {
                int v = Integer.parseInt(val);
                return true;
            } catch (NumberFormatException ex) {
                return false;
            }
        }
        return false;
    }
}
